packages/fast-layouts-react/src/grid/grid.spec.tsx

```diff
@@ -0,0 +1,234 @@
+            );
+
+            expect(rendered.props().style.gridColumn).toBe(2);
+        });
+
+        test("should set an inline style for `gridRow` with a value equal to the `row` prop", () => {
+            const rendered: any = shallow(
+                <Grid row={1} managedClasses={managedClasses} />
+            );
+
+            expect(rendered.props().style.gridRow).toBe(1);
+        });
+
+        test("should set an inline style for alignItems with a value equal to the `verticalAlign` prop", () => {
+            const rendered: any = shallow(
+                <Grid
+                    verticalAlign={GridAlignment.center}
+                    managedClasses={managedClasses}
+                />
+            );
+
+            expect(rendered.props().style.alignItems).toBe(GridAlignment.center);
+        });
+
+        test("should set an inline style for justifyItems with a value equal to the `horizontalAlign` prop", () => {
+            const rendered: any = shallow(
+                <Grid
+                    horizontalAlign={GridAlignment.center}
+                    managedClasses={managedClasses}
+                />
+            );
+
+            expect(rendered.props().style.justifyItems).toBe(GridAlignment.center);
+        });
+
+        test("should set an inline style for gridTemplateColumns with a repeating column count equal to the `columnCount` prop", () => {
+            const rendered: any = shallow(
+                <Grid columnCount={5} managedClasses={managedClasses} />
+            );
+
+            expect(rendered.props().style.gridTemplateColumns).toBe(`repeat(5, 1fr)`);
+        });
+    });
+
+    test("should have a displayName that matches the component name", () => {
+        expect(`${(Grid as any).name}`).toBe(Grid.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(<Grid />);
+        }).not.toThrow();
+    });
+
+    test("should accept unhandledProps", () => {
+        const unhandledProps: GridUnhandledProps = {
+            "aria-hidden": true,
+        };
+
+        const props: GridHandledProps & GridUnhandledProps = {
+            ...unhandledProps,
+        };
+
+        const rendered: any = mount(<Grid managedClasses={managedClasses} {...props} />);
+
+        expect(rendered.first().prop("aria-hidden")).toEqual(true);
+    });
+
+    test("should set a default tag value if no `tag` prop is passed", () => {
+        const rendered: any = mount(<Grid managedClasses={managedClasses} />);
+
+        expect(rendered.props().tag).not.toBe(undefined);
+        expect(rendered.props().tag).toBe(GridTag.div);
+    });
+
+    test("should set a tag value equal to the `tag` prop when passed", () => {
+        const rendered: any = mount(
+            <Grid tag={GridTag.main} managedClasses={managedClasses} />
+        );
+
+        expect(rendered.props().tag).toBe(GridTag.main);
```

Comment:
it would be nice if we tested that the tag was actually used to create a `<main>` element.

Reply:
Sure. Easy enough